aws/data_source_aws_resourcegroupstaggingapi_resources.go

```diff
@@ -0,0 +1,194 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func dataSourceAwsResourceGroupsTaggingAPIResources() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsResourceGroupsTaggingAPIResourcesRead,
+
+		Schema: map[string]*schema.Schema{
+			"exclude_compliant_resources": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"include_compliance_details": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"resource_arn_list": {
+				Type:          schema.TypeSet,
+				Optional:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"filter"},
+			},
+			"resource_type_filters": {
+				Type:          schema.TypeSet,
+				Optional:      true,
+				MaxItems:      100,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"resource_arn_list"},
+			},
+			"filter": {
```

Comment:
To match API naming:

```suggestion
			"tag_filter": {
```